File: novel-library-service/src/test/java/de/roamingthings/tracing/novellibrary/ports/driving/TestDataFactory.java
package de.roamingthings.tracing.novellibrary.ports.driving;

import de.roamingthings.tracing.novellibrary.domain.Novel;

import static de.roamingthings.tracing.novellibrary.CommonTestDataFactory.*;

class TestDataFactory {
    static StoreNovelRequest aStoreNovelRequest() {
        return new StoreNovelRequest(
                NOVEL_AUTHORED,
                NOVEL_TITLE,
                NOVEL_CONTENT
        );
    }

    static Novel aNovel() {
        return new Novel(
                NOVEL_UUID,
                NOVEL_CREATED,
                NOVEL_AUTHORED,
                NOVEL_TITLE,
                NOVEL_CONTENT
        );
    }
}

File: novel-ai-service/src/test/kotlin/de/roamingthings/tracing/novelai/ports/driving/TestDataFactory.kt
package de.roamingthings.tracing.novelai.ports.driving

import de.roamingthings.tracing.novelai.domain.Novel
import de.roamingthings.tracing.novelai.domain.NovelUuid
import java.time.LocalDateTime
import java.time.LocalDateTime.of
import java.time.Month.OCTOBER

val NOVEL_UUID = NovelUuid("ae357ee4-f64d-4995-aba1-474abc81ffe9")
val NOVEL_AUTHORED: LocalDateTime = of(2019, OCTOBER, 19, 23, 42)
const val NOVEL_TITLE = "A Test Novel"
const val NOVEL_CONTENT = "This is a very short test novel."

fun aNovel() = Novel(NOVEL_UUID, NOVEL_AUTHORED, NOVEL_TITLE, NOVEL_CONTENT)
